packages/next/src/server/app-render/collect-segment-data.tsx

```diff
@@ -175,18 +164,16 @@ function collectSegmentDataImpl(
     childAccessTokens[parallelRouteKey] = childAccessToken
   }
 
-  // Spawn a task to render the segment data to a stream.
-  collectedTasks.push(
-    renderSegmentDataToStream(
-      fullPageDataBuffer,
-      clientModules,
-      serverConsumerManifest,
-      segmentPath,
-      segmentPathStr,
-      accessToken,
-      childAccessTokens,
-      segmentBufferMap
-    )
+  // Render the segment data to a stream.
+  await renderSegmentDataToStream(
```

Comment:
Since the token generation has to be async anyway, I removed the parallelization via Promise.all. This isn't a super hot code path and the segment rendering tasks resolve after a microtask so it doesn't really optimize much.